catsim/estimation.py

```diff
@@ -240,3 +241,136 @@ def estimate(
+                      (see: :py:func:`catsim.cat.generate_item_bank`)
+        :param administered_items: a list containing the indexes of items that were already administered
+        :param response_vector: a boolean list containing the examinee's answers to the administered items
+        :param est_theta: a float containing the current estimated proficiency
+        :returns: the current :math:`\\hat\\theta`
+        """
+        if (index is None or self.simulator is None) and (
+            items is None and administered_items is None or response_vector is None or
+            est_theta is None
+        ):
+            raise ValueError(
+                'Either pass an index for the simulator or all of the other optional parameters to use this component independently.'
+            )
+
+        if items is None and administered_items is None and response_vector is None and est_theta is None:
+            items = self.simulator.items
+            administered_items = self.simulator.administered_items[index]
+            response_vector = self.simulator.response_vectors[index]
+            est_theta = self.simulator.latest_estimations[index]
+
+        self._calls += 1
+
+        # need to constrain all estimates between these bounds, rather then, e.g.
+        # min / max difficulties
+        lower_bound, upper_bound = self._bounds
+
+        best_theta = float('-inf')
+        max_ll = float('-inf')
+
+        self._evaluations = 0
+
+        for _ in range(10):
+            intervals = numpy.linspace(lower_bound, upper_bound, 10)
+            if self._verbose:
+                print(('Bounds: ' + str(lower_bound) + ' ' + str(upper_bound)))
+                print(('Interval size: ' + str(intervals[1] - intervals[0])))
+
+            for ii in intervals:
+                self._evaluations += 1
+                ll = irt.log_likelihood(ii, response_vector, items[administered_items]) + numpy.log(norm.pdf(ii, loc = self._prior_mean, scale = self._prior_sd))
+                if ll > max_ll:
+                    max_ll = ll
+
+                    if self._verbose:
+                        print(
+                            (
+                                'Iteration: {0}, Theta: {1}, LL: {2}'.format(
+                                    self._evaluations, ii, ll
+                                )
+                            )
+                        )
+
+                    if abs(best_theta - ii) < float('1e-' + str(self._precision)):
+                        return self._bound_estimate(ii)
+
+                    best_theta = ii
+
+                else:
+                    lower_bound = best_theta - (intervals[1] - intervals[0])
+                    upper_bound = ii
+                    # reset best_theta, in case optimum is to the left of it
+                    max_ll = float('-inf')
+                    break
+
+        return self._bound_estimate(best_theta)
```

Comment:
```suggestion
        return self._bound_estimate(best_theta)
        
```

Adding a newline to the end of the file.